ruby/ql/lib/codeql/ruby/dataflow/internal/SsaImpl.qll

```diff
@@ -421,7 +421,7 @@ import Cached
  *
  * Only intended for internal use.
  */
-class DefinitionExt extends Impl::DefinitionExt {
+deprecated class DefinitionExt extends Impl::DefinitionExt {
```

Comment:
Since these are inside an `internal` folder, and has the `Only intended for internal use` comment, it should be fine to simply delete.

Reply:
They're actually used in a couple of deprecated predicates that ultimately end up in the public `getALastRead` member predicate on `Definition` in the outwards-facing `SSA.qll`, so it's less clear cut. That's why I left them in for now.

Reply:
We can make it private, though, and possibly push in enough casts from `DefinitionExt` to `Definition` that we might be able to eliminate it. Let me check.

Reply:
Right. All the uses of `DefinitionExt` here were actually in contexts for which we could push a context with a cast to `Definition`. So I'll be able to delete this. Commit incoming.